fairscale/nn/data_parallel/sharded_ddp.py

```diff
@@ -4,109 +4,178 @@
 # LICENSE file in the root directory of this source tree.
 
 """
-A distributed data parallel class that works with OSS optimizer.
-
-Adopted from LegacyDistributedDataParallel module from fairseq.
+A module wrapper to go with a Sharded Optimizer in order to handle targeted gradient reduction/gathering automatically.
 """
 
 from contextlib import contextmanager
-import copy
-from typing import Any, Dict, Generator, List, Type, cast
+import logging
+from typing import Any, Callable, Dict, Generator, List, Tuple, Union
 
 import torch
-from torch import Tensor, nn
+from torch import nn
 import torch.distributed as dist
 from torch.nn import Parameter
 
-from fairscale.optim import OSS
+from fairscale.optim.oss import OSS
```

Comment:
ah no idea, it was an autocomplete. The diff compares to the old file but this was written from scratch since different logic, so I did not realize that the import differed. Fixing that